tests/sample/morris/test_morris.py

```diff
@@ -224,3 +222,105 @@ def test_generate_x_star(self):
+
+        expected = problem
+
+        result = define_problem_with_groups(problem)
+
+        assert expected == result
+
+    def test_define_problem_with_groups_no_group_definition(self):
+        """
+        Checks if the function works when the user doesn't define groups at
+        all.
+        """
+        problem = {
+            'num_vars': 8,
+            'names': ['x1', 'x2', 'x3', 'x4', 'x5', 'x6', 'x7', 'x8']}
+
+        expected = {
+            'num_vars': 8,
+            'names': ['x1', 'x2', 'x3', 'x4', 'x5', 'x6', 'x7', 'x8'],
+            'groups': ['x1', 'x2', 'x3', 'x4', 'x5', 'x6', 'x7', 'x8']}
+
+        result = define_problem_with_groups(problem)
+
+        assert expected == result
+
+    def test_define_problem_with_groups_exception(self):
+        """
+        Checks if the function raises an exception when the user makes an
+        inconsistent definition of groups, i.e, only define groups for some
+        variables.
+        """
+        problem = {
+            'num_vars': 8,
+            'names': ['x1', 'x2', 'x3', 'x4', 'x5', 'x6', 'x7', 'x8'],
+            'groups': ['G1', 'G1', 'G1', 'G2', 'G2', 'G2']}
+
+        with raises(ValueError):
+            define_problem_with_groups(problem)
+
+    def test_check_if_num_levels_is_even_check_odd(self):
+        """
+        Checks if a warn is raised when the number of tests is odd
+        """
+        with warns(None) as record:
+            check_if_num_levels_is_even(5)
+
+        assert record
+
+    def test_check_if_num_levels_is_even_check_even(self):
+        """
+        Checks if a warn is not raised when the number of tests is even.
+        """
+        with warns(None) as record:
+            check_if_num_levels_is_even(4)
+
+        assert not record
+
+    @mark.xfail()
+    def test_check_group_membership_all_ok(self):
+        """
+        Checks if no errors are raised when group_membership is defined
+        correctly. This test is expected to fail, and the message XFAIL should
+        be displayed by pycharm.
+        """
+        # Creates a dummy variable
+        group_membership = np.empty((3, 3), dtype=np.int)
+
+        with raises(ValueError):
+            assert check_group_membership(group_membership)
+
+        with raises(TypeError):
+            assert check_group_membership(group_membership)
+
+    def test_check_group_membership_type_error(self):
+        """
+        Checks if an error is raised if group_membership has a wrong type.
+        """
+        # Creates a dummy variable
+        group_membership = 1.0
+
+        with raises(TypeError):
+            assert check_group_membership(group_membership)
+
+    def test_check_group_membership_value_error(self):
+        """
+        Checks if an error is raised if group_membership has None type.
+        """
+        # Creates a dummy variable
+        group_membership = None
+
+        with raises(ValueError):
+            assert check_group_membership(group_membership)
```

Comment:
Add newline at end of file to conform to PEP8.

These tests are excellent and worthwhile additions, thanks!